### ISSUE TITLE
Support vault integration

### ISSUE THREAD

**Issue by fairbairn** [hashicorp/terraform-aws-nomad#69]
run-nomad does not currently support the vault stanza for vault integration with server or client

This would add

vault {
  enabled
  address
  create_from_role
  token
}

at a minimum.

**Comment by brikis98** [comment]
Any reason you can't specify this stanza, and all other custom configs, as specified in the [overriding the configuration docs](https://github.com/hashicorp/terraform-aws-nomad/tree/master/modules/run-nomad#overriding-the-configuration)?

**Comment by fairbairn** [comment]
We can.

**Comment by brikis98** [comment]
I don't think it would be maintainable for `run-nomad` to support every parameter. It should support the ones that affect, say, 80-90% of users, and for the rest, using a custom config file and _then_ calling `run-nomad` is the recommended way to go. You can package the custom config file with your AMI. If there is some data that must be passed in at run-time, you can pass that in via User Data.
